### PR TITLE
refactor(supernova): introducing ReactQuery

### LINKED ISSUE #628: [TASK] (supernova): Remove Alerts Worker and Introduce React Query
## Description
Replace the alerts worker with React Query while maintaining the existing functionality without modifying the store and other components. This refactor is a preparatory step for a future rearchitecture and will enable fetching with filters in a subsequent phase.

## Steps
1. **Remove Alerts Worker**
   - Remove the current implementation of the alerts worker used for fetching data.

2. **Introduce React Query**
   - Use React Query to fetch alerts data.
   - Ensure the fetching logic retains the same interval as the alerts worker.

3. **Migrate Alert Worker Logic to React Query**
   - Transfer the logic from the worker function into the React Query fetching promise.
   - Ensure the fetched data structure matches the existing structure used in the application.

4. **Save Fetching Results**
   - Save the results fetched by React Query into the Zustand store.
   - Maintain compatibility with the existing store structure and ensure no changes are required in other components.

## Acceptance Criteria
- The alerts worker is fully removed.
- React Query is successfully integrated for fetching alerts data.
- The fetching interval remains consistent with the worker's behavior.
- The results fetched by React Query match the existing data structure and are stored in the Zustand store.
- All existing components relying on the Zustand store remain functional without modification.

## Related to
- #621

## Notes
- This refactor does not introduce new functionality, such as filtering, but sets the groundwork for such changes in a subsequent iteration.

### LINKED ISSUE #659: [TASK] (supernova): Remove Silences Worker and Introduce React Query
## Description
Replace the silences worker with React Query while maintaining the existing functionality without modifying the store and other components. This refactor is a preparatory step for a future rearchitecture.

## Steps
1. **Remove Silences Worker**
   - Remove the current implementation of the silences worker used for fetching data.

2. **Introduce React Query**
   - Use React Query to fetch alerts data.
   - Ensure the fetching logic retains the same interval as the alerts worker.

3. **Migrate Alert Worker Logic to React Query**
   - Transfer the logic from the worker function into the React Query fetching promise.
   - Ensure the fetched data structure matches the existing structure used in the application.

4. **Save Fetching Results**
   - Save the results fetched by React Query into the Zustand store.
   - Maintain compatibility with the existing store structure and ensure no changes are required in other components.

## Acceptance Criteria
- The silences worker is fully removed.
- React Query is successfully integrated for fetching silences data.
- The fetching interval remains consistent with the worker's behavior.
- The results fetched by React Query match the existing data structure and are stored in the Zustand store.
- All existing components relying on the Zustand store remain functional without modification.

## Related to
- #621

## Notes
- This refactor does not introduce new functionality, such as filtering, but sets the groundwork for such changes in a subsequent iteration.

### PR DESCRIPTION
**Pull request by TilmanHaupt**
# Summary

Closes #628 
Closes #659 

- The alerts worker is removed.
- The silences worker is removed.
- React Query is successfully integrated for fetching alerts data.
- Generalized Bounding for React Query is successfully integrated for fetching alerts data
- The fetching interval remains consistent with the worker's behavior.
- The results fetched by React Query match the existing data structure and are stored in the Zustand store.
- All existing components relying on the Zustand store remain functional without modification.
- Loading and error states are now deleted and we now use reactQuery states
- Errors for Alerts are now show in messages.

# Testing Instructions

1. `npm i`
2. `npm run TASK`

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

### CODE REVIEW

File: apps/supernova/src/App.jsx [hunk 1]
```diff
@@ -9,12 +9,13 @@ import { AppShellProvider, CodeBlock } from "@cloudoperators/juno-ui-components"
 import AppContent from "./AppContent"
 import styles from "./styles.scss?inline"
 import { StoreProvider } from "./components/StoreProvider"
-import AsyncWorker from "./components/AsyncWorker"
```
Comment (andypf) [inline apps/supernova/src/App.jsx:12]: Maybe we don't need AsynchWorker at all.

File: apps/supernova/src/AppContent.jsx [hunk 2]
```diff
@@ -48,6 +46,15 @@ const AppContent = () => {
     })
   }, [silencesError])
 
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/AppContent.jsx:49]: Try to just remove useEffect:
```jsx
if (alertsError) {
  addMessage({
     variant: "error",
     text: parseError(alertsError),
   })
}
    
```

File: apps/supernova/src/AppContent.jsx [hunk 3]
```diff
@@ -56,33 +63,15 @@ const AppContent = () => {
...
-          {isSilencesLoading ? (
-            <Stack gap="2">
-              <span>Loading</span>
-              <Spinner variant="primary" />
-            </Stack>
-          ) : (
-            <SilencesList />
-          )}
+          <>
+            <PredefinedFilters />
```
Comment (andypf) [inline apps/supernova/src/AppContent.jsx:67]: Ensure that predefinedFilter also implements a loading indicator.

File: apps/supernova/src/components/AsyncWorker.jsx [hunk 4]
```diff
@@ -3,11 +3,9 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import useAlertmanagerAPI from "../hooks/useAlertmanagerAPI"
 import useUrlState from "../hooks/useUrlState"
 
-const AsyncWorker = ({ endpoint }) => {
-  useAlertmanagerAPI(endpoint)
+const AsyncWorker = () => {
   useUrlState()
```
Comment (andypf) [inline apps/supernova/src/components/AsyncWorker.jsx:9]: you probably can move useUrlState to the main component App

File: apps/supernova/src/components/alerts/AlertsList.jsx [hunk 5]
```diff
@@ -3,18 +3,55 @@
...
+  const { setAlertsData, setError, setIsLoading } = useAlertsActions()
+
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["alerts", endpoint], // Pass the query key as part of the object
+    queryFn: () => fetchAlerts(endpoint), // Pass the query function
+    enabled: !!endpoint, // Add any other options here
+    refetchInterval: 5 * 60 * 1000, // 5 Min
+  })
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/alerts/AlertsList.jsx:38]: try to remove useEffect.

File: apps/supernova/src/components/alerts/AlertsList.jsx [hunk 6]
```diff
@@ -3,18 +3,55 @@
...
+    if (data) {
+      setAlertsData({ items: data.alerts, counts: data.counts })
+    }
+  }, [data])
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/alerts/AlertsList.jsx:44]: same

File: apps/supernova/src/components/alerts/AlertsList.jsx [hunk 7]
```diff
@@ -3,18 +3,55 @@
...
+    if (error) {
+      setError(error)
+    }
+  }, [error])
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/alerts/AlertsList.jsx:50]: same

File: apps/supernova/src/components/silences/SilencesList.jsx [hunk 8]
```diff
@@ -38,11 +40,46 @@ my-px
...
+  } = useSilencesActions()
+
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["silences", endpoint], // Pass the query key as part of the object
+    queryFn: () => fetchSilences(endpoint), // Pass the query function
+    enabled: !!endpoint, // Add any other options here
+    refetchInterval: 5 * 60 * 1000, // 5 Min
+  })
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/silences/SilencesList.jsx:62]: please don't use useEffect here

File: apps/supernova/src/components/silences/SilencesList.jsx [hunk 9]
```diff
@@ -38,11 +40,46 @@ my-px
...
+    if (data) {
+      setSilences({
+        items: data?.silences,
+        itemsHash: data?.silencesHash,
+        itemsByState: data?.silencesBySate,
+      })
+    }
+  }, [data])
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/silences/SilencesList.jsx:72]: same

File: apps/supernova/src/components/silences/SilencesList.jsx [hunk 10]
```diff
@@ -38,11 +40,46 @@ my-px
...
+    if (error) {
+      setSilencesError(error)
+    }
+  }, [error])
+
+  useEffect(() => {
```
Comment (andypf) [inline apps/supernova/src/components/silences/SilencesList.jsx:78]: same

**Review by andypf — COMMENTED** [review]
Great Job, but pleas address my comments.

**Review by barsukov — COMMENTED** [review]
This is a great work, I agree with everything what @andypf said so please fix the issues and we could merge it.

File: apps/supernova/src/App.jsx [hunk 11]
```diff
@@ -38,8 +39,9 @@ function App(props = {}) {
     <MessagesProvider>
       <CustomAppShell>
         <ErrorBoundary fallbackRender={fallbackRender}>
```
Comment (ArtieReus) [inline apps/supernova/src/App.jsx:41]: [next PR]: consider moving up the `ErrorBoundary` to catch errors within the CustomAppShell and MessagesProvider

File: apps/supernova/src/workers/alerts.js [hunk 12]
```diff
@@ -3,20 +3,25 @@
...
 
-/**
- * @param {string} endpoint
- * @param {object} options
- * @returns {function} fetch function
- */
-const fetchAction = (endpoint, options = {}) => {
-  return get(`${endpoint}/alerts`, { params: options.params }).then((items) => {
-    // sort alerts
+export const fetchAlerts = async (endpoint) => {
```
Comment (ArtieReus) [inline apps/supernova/src/workers/alerts.js:9]: I would recommed to setup react Query with all queries and functions in a central place  based on the idea in heureka:
https://github.com/cloudoperators/juno/blob/b007ad29c901861f480078f5b6f64e933c482a19/apps/heureka/src/hooks/useQueryClientFn.js#L20

File: apps/supernova/src/workers/alerts.js
Comment (ArtieReus) [inline apps/supernova/src/workers/alerts.js:1]: I would recomend to move functions to the lib folder

File: apps/supernova/src/workers/silences.js
Comment (ArtieReus) [inline apps/supernova/src/workers/silences.js:1]: same as alert worker

File: apps/supernova/src/components/silences/SilencesList.jsx [hunk 13]
```diff
@@ -38,11 +40,46 @@ my-px
...
+  const {
+    setSilences,
+    setIsUpdating: setSilencesIsUpdating,
+    setError: setSilencesError,
+    setSilencesStatus,
+    setSilencesRegEx,
+  } = useSilencesActions()
+
+  const { data, error, isLoading } = useQuery({
+    queryKey: ["silences", endpoint], // Pass the query key as part of the object
```
Comment (ArtieReus) [inline apps/supernova/src/components/silences/SilencesList.jsx:56]: can be simplified when using predefined queries as described in the comment below within the alert worker file

**Review by ArtieReus — CHANGES_REQUESTED** [review]
As described in the comments

File: apps/supernova/src/App.jsx (hunk 11 above)
Comment (TilmanHaupt) [reply apps/supernova/src/App.jsx:41]: resolved

**Review by barsukov — APPROVED** [review]

**Review by ArtieReus — APPROVED** [review]
Looks great.  Really good job. You improved the code!! Tests in the preview looks good to me. Thank you

**Pull request merged**